### ISSUE TITLE
[BUG] documentSymbol does not handle Go FQN patterns

### ISSUE THREAD

**Issue by savitharaghunathan** [konveyor/analyzer-lsp#1102]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Undefined (Default)

## Problem

  The `documentSymbol` fallback path in the generic service client does not handle Go FQN patterns (e.g.,
  `v1beta1.CustomResourceDefinition`). This causes inconsistent behavior between kantra and kai:

  - **kantra**: Rules with Go FQN patterns flag correctly (uses `workspace/symbol` from gopls directly)
  - **kai**: Same rules produce no matches (the Go VSCode proxy intentionally returns `[]` for
  `workspace/symbol`, forcing the `documentSymbol` fallback which uses the default `SymbolSearchHelper`)

 

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by shawn-hurley** [comment]
I think that we should chalk this up to another reason to implement #1013. The generic approach just doesn't work, there is too much difference between language servers
